app/models/subject.rb

```diff
@@ -123,6 +123,21 @@ def update_comments
     end
   end
 
+  def comment(user, comment_body)
+    return if comment_body.nil? || comment_body.empty?
+    comment = comments.create(author: user.github_login, body: comment_body)
+    CommentWorker.perform_async_if_configured(comment.id, user.id, self.id)
+  end
+
+  def comment_on_github(comment, user)
+    return if comment.body.empty?
+    remote_comment = user.github_client.post "#{url}/comments", {body: comment.body}
```

Comment:
The `github_client` use here needs to be swapped out with the particular kind of client depending on the permissions setup for that repository. Namely, if the repository is private and has the github app installed, then the `app_installation_client` needs to be used instead of the regular `github_client`.